kie-server-parent/kie-server-services/kie-server-services-jbpm-queries/src/main/java/org/kie/server/services/jbpm/queries/RequestCallback.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm.queries;
+
+import org.kie.server.jbpm.queries.api.model.definition.BaseQueryFilterSpec;
+import org.kie.server.services.jbpm.queries.util.QueryStrategy;
+
+/**
+ * Provides the context for query request execution, including things like payload and unmarshalled {@link BaseQueryFilterSpec}. 
+ * 
+ * 
+ * @author <a href="mailto:duncan.doyle@redhat.com">Duncan Doyle</a>
+ */
+public interface RequestCallback {
```

Comment:
is this interface and then specific impl really needed? can't we simply add argument of type BaseQueryFilterSpec?